flink-clients/src/main/java/org/apache/flink/client/program/JobWithJars.java

```diff
@@ -122,7 +123,11 @@ public static void checkJarFile(URL jar) throws IOException {
 		if (!jarFile.canRead()) {
 			throw new IOException("JAR file can't be read '" + jarFile.getAbsolutePath() + "'");
 		}
-		// TODO: Check if proper JAR file
+		try (JarFile f = new JarFile(jarFile)) {
+
+		} catch (IOException e) {
+			throw new IOException("Error while opening jar file '" + jarFile.getAbsolutePath() + "'", e);
+		}
```

Comment:
could we replace the first check where we create the `File` with this one?

Reply:
Good suggestion, I have updated this PR, please help take a look, thanks a lot!